packages/common/src/utils/formatting.ts

```diff
@@ -271,13 +275,82 @@ export function formatFieldValue(
     }
 }
 
+export function formatNumberWithSeparator(
+    value: number,
+    separator: NumberSeparator = NumberSeparator.COMMA_PERIOD,
+    round: number = 0,
+): string {
+    const options =
+        round <= 0
+            ? {
+                  maximumSignificantDigits: Math.max(
+                      Math.floor(value).toString().length + round,
+                      1,
+                  ),
+                  maximumFractionDigits: 0,
+              }
+            : {
+                  maximumFractionDigits: Math.min(round, 20),
+                  minimumFractionDigits: Math.min(round, 20),
+              };
+
+    switch (separator) {
+        case NumberSeparator.COMMA_PERIOD:
+            return value.toLocaleString('en-US', options);
+        case NumberSeparator.SPACE_PERIOD:
+            return value.toLocaleString('en-US', options).replace(/,/g, ' ');
+        case NumberSeparator.PERIOD_COMMA:
+            return value.toLocaleString('de-DE', options);
+        case NumberSeparator.NO_SEPARATOR_PERIOD:
+            return value.toLocaleString('en-US', {
+                ...options,
+                useGrouping: false,
+            });
+        default:
+            return assertUnreachable(separator, 'Unknown separator');
+    }
+}
+export function formatTableCalculationValue(
+    field: TableCalculation,
+    value: unknown,
+): string {
+    if (field.format?.type === undefined) return formatValue(value);
+    switch (field.format.type) {
+        case TableCalculationFormatType.DEFAULT:
+            return formatValue(value);
+
+        case TableCalculationFormatType.PERCENT:
+            if (valueIsNaN(value)) {
+                return `${value}`;
+            }
+            const formatted = formatNumberWithSeparator(
+                Number(value) * 100,
+                field.format.separator,
+                field.format.round,
+            );
+            return `${formatted}%`;
+
+        default:
+            return assertUnreachable(
+                field.format.type,
+                `Table calculation format type ${field.format.type} is not valid`,
+            );
+    }
+}
+
 export function formatItemValue(
     item: Field | AdditionalMetric | TableCalculation | undefined,
     value: unknown,
     convertToUTC?: boolean,
 ): string {
     if (value === null) return '∅';
     if (value === undefined) return '-';
+    if (
+        item !== undefined &&
+        !isAdditionalMetric(item) &&
+        isTableCalculation(item)
+    )
+        return formatTableCalculationValue(item, value);
```

Comment:
Can we use brackets `{}` in this if block? was hard to tell the difference between the two returns